### PR TITLE
Allow local storage decorator to register as property

### PR DESCRIPTION
**Pull request by bramkragten**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/common/decorators/local-storage.ts [hunk 1]
```diff
@@ -32,9 +33,13 @@ class Storage {
 
 const storage = new Storage();
 
-export const LocalStorage = (key?: string) => {
-  return (element: ClassElement, propName: string) => {
-    const storageKey = key || propName;
+export const LocalStorage = (
```
Comment (balloob) [inline src/common/decorators/local-storage.ts:36]: Should we listen to the `storage` event? 

https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event

**Review by balloob — APPROVED** [review]

File: src/common/decorators/local-storage.ts (hunk 1 above)
Comment (bramkragten) [reply src/common/decorators/local-storage.ts:36]: For when it is changed outside of the class? Would be nice, but we don't have a use case now I think?

Comment (balloob) [reply src/common/decorators/local-storage.ts:36]: Two windows open. Changes to one are applied to the other

Comment (bramkragten) [reply src/common/decorators/local-storage.ts:87]: Added

**Pull request merged**